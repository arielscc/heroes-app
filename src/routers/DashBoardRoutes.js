import { Redirect, Route, Switch } from 'react-router-dom';
import { DcScreen } from '../components/dc/DcScreen';
import { HeroScreen } from '../components/heroes/HeroScreen';
import { LoginScreen } from '../components/login/LoginScreen';
import { MarvelScreen } from '../components/marvel/MarvelScreen';
import { Navbar } from '../components/ui/Navbar';

export const DashBoardRoutes = () => {
  return (
    <>
      <Navbar />

      <Switch>
        <Route exact path="/marvel" component={MarvelScreen} />
        <Route exact path="/hero/:heroeId" component={HeroScreen} />
        <Route exact path="/dc" component={DcScreen} />
        <Route exact path="/login" component={LoginScreen} />

        <Redirect to="/marvel" />
      </Switch>
    </>
  );
};
