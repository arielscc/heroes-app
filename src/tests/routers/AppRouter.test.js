import { mount } from 'enzyme';
import { useReducer } from 'react';
import { AuthContext } from '../../auth/AuthContext';
import { authReducer } from '../../auth/authReducer';
import { AppRouter } from '../../routers/AppRouter';

describe('Pruebas en <AppRouter/>', () => {
  const valueContext = {
    user: {
      logged: false,
    },
    dispatch: jest.fn(),
  };

  test('debe de mostrar el login si no esta autenticado', () => {
    const wrapper = mount(
      <AuthContext.Provider value={valueContext}>
        <AppRouter />
      </AuthContext.Provider>
    );

    expect(wrapper).toMatchSnapshot();
  });

  test('debe de mostrar el componente marvel, si estÃ¡ autenticado', () => {
    const valueContext = {
      user: {
        logged: true,
        name: 'Ariel',
      },
      dispatch: jest.fn(),
    };
    const wrapper = mount(
      <AuthContext.Provider value={valueContext}>
        <AppRouter />
      </AuthContext.Provider>
    );
    expect(wrapper.contains('Ariel')).toBe(true);
  });
});
