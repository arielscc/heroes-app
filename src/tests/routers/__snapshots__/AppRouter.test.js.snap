// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter/> debe de mostrar el login si no esta autenticado 1`] = `
<div
  className="flex justify-between m-10"
>
  <h1>
    LoginScreen
  </h1>
  <button
    className="py-2 px-4 font-semibold rounded-lg shadow-md text-white bg-pink-600 hover:bg-pink-700 focus:outline-none"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
